`~/.aider.conf.yml` affects test suite

### Issue

Defining `autocommits: false` in `~/.aider.conf.yml` causes `tests/basic/test_main.py:TestMain.test_main_args` to fail. It seems the test isn't properly isolated from the host filesystem:

`~/.aider.conf.yml`:
```yaml
auto-commits: false
```
`$ pytest --ff tests/basic/test_main.py -k test_main_args`:
```python
================================ test session starts =================================
platform linux -- Python 3.12.4, pytest-8.3.2, pluggy-1.5.0
rootdir: /home/akaihola/repos/ai/aider-chat
configfile: pytest.ini
plugins: anyio-4.4.0
collected 38 items / 37 deselected / 1 selected                                      
run-last-failure: rerun previous 1 failure first

tests/basic/test_main.py F                                                     [100%]

====================================== FAILURES ======================================
______________________________ TestMain.test_main_args _______________________________

self = <test_main.TestMain testMethod=test_main_args>

    def test_main_args(self):
        with patch("aider.coders.Coder.create") as MockCoder:
            # --yes will just ok the git repo without blocking on input
            # following calls to main will see the new repo already
            main(["--no-auto-commits", "--yes"], input=DummyInput())
            _, kwargs = MockCoder.call_args
            assert kwargs["auto_commits"] is False
    
        with patch("aider.coders.Coder.create") as MockCoder:
            main(["--auto-commits"], input=DummyInput())
            _, kwargs = MockCoder.call_args
            assert kwargs["auto_commits"] is True
    
        with patch("aider.coders.Coder.create") as MockCoder:
            main([], input=DummyInput())
            _, kwargs = MockCoder.call_args
            assert kwargs["dirty_commits"] is True
>           assert kwargs["auto_commits"] is True
E           assert False is True

tests/basic/test_main.py:153: AssertionError
-------------------------------- Captured stdout call --------------------------------
Git repository created in the current working directory.
Added .aider* to .gitignore

Use /help <question> for help, run "aider --help" to see cmd line args

Use /help <question> for help, run "aider --help" to see cmd line args

Use /help <question> for help, run "aider --help" to see cmd line args
============================== short test summary info ===============================
FAILED tests/basic/test_main.py::TestMain::test_main_args - assert False is True
========================== 1 failed, 37 deselected in 3.02s ==========================
```

### Version and model info

Aider: 0.52.2-dev 6fb3416757a7860bcf4daaa4c4c2249fd2ac305c

--- a/tests/basic/test_main.py
+++ b/tests/basic/test_main.py
@@ -26,10 +26,14 @@ class TestMain(TestCase):
         self.tempdir_obj = IgnorantTemporaryDirectory()
         self.tempdir = self.tempdir_obj.name
         os.chdir(self.tempdir)
+        # Fake home directory prevents tests from using the real ~/.aider.conf.yml file:
+        self.homedir_obj = IgnorantTemporaryDirectory()
+        os.environ["HOME"] = self.homedir_obj.name
 
     def tearDown(self):
         os.chdir(self.original_cwd)
         self.tempdir_obj.cleanup()
+        self.homedir_obj.cleanup()
         os.environ.clear()
         os.environ.update(self.original_env)
 

